server/game/gameSystems/MoveCardSystem.ts

```diff
@@ -5,7 +5,7 @@ import * as EnumHelpers from '../core/utils/EnumHelpers';
 import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
 
 export interface IMoveCardProperties extends ICardTargetSystemProperties {
-    destination?: Location;
+    destination?: Location.Discard | Location.Deck;
```

Comment:
This feels odd to me. If this system is only for a subset of card moves it should probably have a different name.